RecoPixelVertexing/PixelTriplets/plugins/CAGraph.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * CAGraph.h
+ *
+ *  Created on: Aug 19, 2016
+ *      Author: fpantale
+ */
+
+#ifndef CAGRAPH_H_
+#define CAGRAPH_H_
+
+#include <vector>
+#include <array>
+#include <string>
+#include <queue>
+#include <functional>
+#include "CACell.h"
+
+class CALayer
+{
+public:
+	CALayer(const std::string& layerName, std::size_t numberOfHits )
+	: theName(layerName)
+	{
+		isOuterHitOfCell.resize(numberOfHits);
+	}
+
+	bool operator==(const std::string& otherString)
+	{
+		return otherString == theName;
+	}
+
+	std::string name() const
+	{
+		return theName;
+	}
+
+	std::vector<int> theOuterLayerPairs;
+	std::vector<int> theInnerLayerPairs;
+
+	std::vector<int> theOuterLayers;
+	std::vector<int> theInnerLayers;
+	std::vector< std::vector<CACell*> >  isOuterHitOfCell;
```

Comment:
what's a good reason to have public data members in this class?
Maybe just change to a struct (a struct can not have any complicated functions though)